DNN Platform/Tests/DotNetNuke.Tests.Web/InternalServices/TabVersionControllerTests.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using NUnit.Framework;
+using Moq;
+
+using DotNetNuke.Common.Utilities;
+using DotNetNuke.Entities.Tabs.TabVersions;
+using DotNetNuke.Tests.Utilities.Mocks;
+using DotNetNuke.Entities.Users;
+using DotNetNuke.Entities.Controllers;
+using DotNetNuke.Common;
+
+namespace DotNetNuke.Tests.Web.InternalServices
+{
+    [TestFixture]
+    class TabVersionControllerTests
+    {
+
+        private class TabVersionControllerTestable : TabVersionController
+        {
+            protected override TimeZoneInfo GetDatabaseDateTimeOffset()
+            {
+                var timeZoneId = "UTC";
+                return TimeZoneInfo.CreateCustomTimeZone(timeZoneId, new TimeSpan(0, 0, 0), timeZoneId, timeZoneId);
+            }
+        }
+
+        private Mock<ICBO> _mockCBO;
+        private Mock<IUserController> _mockUserController;
+        private Mock<IHostController> _mockHostController;
+
+        private const int UserID = 1;
+        private const int TabID = 99;
+
+        private int OffsetHours { get; set; }
+        
+        // Assuming 12:00 Aug 15, 2018 server local time
+        private readonly DateTime ServerCreateOnDate = new DateTime(2018, 08, 15, 12, 0, 0, DateTimeKind.Unspecified);
+        
+
+        [SetUp]
+        public void Setup()
+        {
+            MockComponentProvider.ResetContainer();
+            SetupCBO();
+            SetupHostController();
+        }
+
+        [Test]
+        [TestCase(-11, "2018-08-15 1:00:00.0")]
+        [TestCase(-2, "2018-08-15 10:00:00.0")]
+        [TestCase(12, "2018-08-16 0:00:00.0")]
+        public void GetTabVersions_Verify_User_Preferred_TimeZone(int offsetHours, string expectedDateTimeString)
+        {
+
+            // Arrange
+            OffsetHours = offsetHours;
+            SetupUserController();
+
+            // Act
+            var tabVersionController = new TabVersionControllerTestable();
+            var tabVersions = tabVersionController.GetTabVersions(TabID);
+            var tabVersion = tabVersions.FirstOrDefault();
+            var expectedDate = DateTime.Parse(expectedDateTimeString);
```

Comment:
Parsing datetimes from string is prone to fail on machines with different locale.

Please consider replacing `[TestCase]` with [`[TestCaseSource]`](https://github.com/nunit/docs/wiki/TestCaseSource-Attribute) where you would create appropriate values with `new DateTime()`